analyzer/src/analyze/condition.py

```diff
@@ -11,6 +17,16 @@ def __init__(
         self.sql_column = sql_column
         self.value = value
 
-    def check(self, data):
+    def check(self, data: Dict[DataDictKey, DataDictValue]):
         data_value = data[(CONDITION_FLAG, (self.sql_column.table, self.sql_column.column))]
-        return (self.action == "EXCLUDE") ^ (self.value == data_value)
+
+        # if data_value is an array, all of its values should be similar
+        if isinstance(data_value, tuple):
+            if not all(el == data_value[0] for el in data_value[1:]):
+                raise ValueError(
+                    "Conditions can only be checked against arrays with identical values."
+                )
+            data_value = data_value[0]
+
+        # TODO handle types
```

Comment:
does that mean we don't handle booleans yet for instance ?

Reply:
I think so yes 🙁